usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -381,45 +381,82 @@ def apply_annotations_to_sql(raw_query, aliases):
     want to use the efficiency of psql's COPY method and keep the column names, we need to allow these scenarios. This
     function simply outputs a modified raw sql which does the aliasing, allowing these scenarios.
     """
-    aliases_copy = list(aliases)
 
-    # Extract everything between the first SELECT and the last FROM
-    query_before_group_by = raw_query.split("GROUP BY ")[0]
-    query_before_from = re.sub("SELECT ", "", " FROM".join(re.split(" FROM", query_before_group_by)[:-1]), count=1)
+    select_statements = _select_columns(raw_query)
 
+    DIRECT_SELECT_QUERY_REGEX = r'^[^ ]*\."[^"]*"$'  # Django is pretty consistent with how it prints out queries
     # Create a list from the non-derived values between SELECT and FROM
-    selects_str = re.findall(
-        r"SELECT (.*?) (CASE|CONCAT|SUM|COALESCE|STRING_AGG|MAX|EXTRACT|\(SELECT|FROM)", raw_query
-    )[0]
-    just_selects = selects_str[0] if selects_str[1] == "FROM" else selects_str[0][:-1]
-    selects_list = [select.strip() for select in just_selects.strip().split(",")]
+    selects_list = list(filter(lambda str: re.search(DIRECT_SELECT_QUERY_REGEX, str.strip()), select_statements))
 
     # Create a list from the derived values between SELECT and FROM
-    remove_selects = query_before_from.replace(selects_str[0], "")
-    deriv_str_lookup = re.findall(
-        r"(CASE|CONCAT|SUM|COALESCE|STRING_AGG|MAX|EXTRACT|\(SELECT|)(.*?) AS (.*?)( |$)", remove_selects
-    )
+    aliased_list = list(filter(lambda str: not re.search(DIRECT_SELECT_QUERY_REGEX, str.strip()), select_statements))
     deriv_dict = {}
-    for str_match in deriv_str_lookup:
-        # Remove trailing comma and surrounding quotes from the alias, add to dict, remove from alias list
-        alias = str_match[2][:-1].strip() if str_match[2][-1:] == "," else str_match[2].strip()
-        if (alias[-1:] == '"' and alias[:1] == '"') or (alias[-1:] == "'" and alias[:1] == "'"):
-            alias = alias[1:-1]
-        deriv_dict[alias] = "{}{}".format(str_match[0], str_match[1]).strip()
-        # Provides some safety if a field isn't provided in this historical_lookup
-        if alias in aliases_copy:
-            aliases_copy.remove(alias)
-
-    # Validate we have an alias for each value in the SELECT string
-    if len(selects_list) != len(aliases_copy):
-        raise Exception("Length of aliases doesn't match the columns in selects")
+    for str in aliased_list:
+        split_string = _top_level_split(str, " AS ")
+        alias = split_string[1].replace('"', "").replace(",", ",").strip()
```

Comment:
what is `.replace(",", ",")` for?

Reply:
Supposed to be `replace(",","")`. Good catch